src/App.js

```diff
@@ -43,11 +44,19 @@ export default function App() {
   const ErrorHandler = useContext(ErrorContext)
   const { errorTitle, errorMessage, setError } = useError()
 
-  useEffect(() => {
-    const socket = io('http://localhost:5001', {
-      withCredentials: true
-    })
-  }, [])
+  const [socket, setSocket] = useState();
+
+    useLayoutEffect(() => {
+        const fetchSocket = async () => {
+            const openSocket = await io('http://localhost:5001', {
+                withCredentials: true
+            })
+            await setSocket(openSocket);
+            console.log('Socket connected');
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/OrangeMGE-ITP/SmartQ-ITP-Front/pullRequest?prid=8748018)
```suggestion
            console.log("Socket connected");
```